modules/standard/Lists.chpl

```diff
@@ -0,0 +1,363 @@
+    */
+    proc init=(other: List) {
+      this._etype = other._etype;
+    }
+  
+    proc deinit() {}
+
+    /*
+      Add an element to the end of this List.
+
+      .. note::
+
+        Appending an element to a List will never invalidate references to
+        elements in the List.
+
+      :arg x: An element to append.
+    */
+    proc append(x: this._etype) {}
+
+    /*
+      Extend this List by appending all the elements from another List.
+
+      :arg other: A List, the elements of which are appended to this List.
+    */
+    proc extend(other: List(this._etype)) {}
+
+    /*
+      Extend this List by appending all the elements from an array.
+
+      :arg other: An array, the elements of which are appended to this List.
+    */
+    proc extend(other: this._etype[?d]) {}
+
+    /*
+      Insert an element at a given position in this List, shifting all elements
+      following that index one to the right. Note that `a.insert(a.size, x)`
+      is equivalent to `a.append(x)`.
+
+      :arg i: The index of the element at which to insert.
+      :arg x: The element to insert.
+
+      :throws IllegalArgumentError: If the given index is out of bounds.
+    */
+    proc insert(i: int, x: this._etype) throws {}
+
+    /*
+      Remove the first item from this List whose value is equal to _x_.
+
+      :arg x: The element to remove.
+
+      :throws IllegalArgumentError: If there is no such item.
+    */
+    proc remove(x: this._etype) throws {}
+
+    /*
+      Remove the item at the given position in this List, and return it. If no
+      index is specified, remove and return the last item in this List.
+
+      :arg i: The index of the element to remove. Defaults to the last item
+        in this List, or `a.remove(a.size - 1)`.
+      
+      :return: The item removed.
+
+      :throws IllegalArgumentError: If the given index is out of bounds.
+    */
+    proc pop(i: int=this._size - 1): this._etype throws {
+      return nil;
+    }
+
+    /*
+      Clear the contents of this List.
+    */
+    proc clear() {}
+
+    /*
+      Return a zero-based index into this List of the first item whose value
+      is equal to _x_.
+
+      :arg x: An element to search for.
+
+      :return: The index of the element to search for.
+
+      :throws IllegalArgumentError: If the given element cannot be found.
+    */
+    proc indexOf(x: _etype, start: int=0, end: int=_size): int throws {
+      return 0; 
+    }
+
+    /*
+      Return the number of times _x_ appears in this List.
+
+      :arg x: An element to count.
+
+      :return: The number of times a given element is found in this List.
+      :rtype: `int`
+    */
+    proc count(x: this._etype): int {
+      return 0;
+    }
+
+    /*
+      Sort the items of this List in place. The parameters of this method are
+      holdovers from Python. This method could more closely integrate with
+      Chapel sorting APIs.
+
+      :arg key: Unused in current API.
```

Comment:
If this is unused, why is it here?  As an alternative?

Reply:
This is one of the methods that I carried over from the Python API, and I decided to leave it in as a point of debate about how we can integrate our containers with the sorting APIs (or what that looks like).

Reply:
You'd just make a totally generic `comparator` argument instead of `key`. The `reverse` argument is also not necessary.

Reply:
Ok, so following along with what is documents in `modules/packages/Sort.chpl`, then. The reverse argument is part of the Python API, you would prefer to drop that in favor of just calling `reverse` instead?

Reply:
Nope, drop it because we have ReverseComparator to do that